packages/interface/src/components/complementary-area/index.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { Animate, Button, Panel, Slot, Fill } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { withPluginContext } from '@wordpress/plugins';
+import { starEmpty, starFilled } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import ComplementaryAreaHeader from '../complementary-area-header';
+import PinnedItems from '../pinned-items';
+
+function ComplementaryAreaSlot( { scope, ...props } ) {
+	return <Slot name={ `ComplementaryArea/${ scope }` } { ...props } />;
+}
+
+function ComplementaryAreaFill( { scope, children, className } ) {
+	return (
+		<Fill name={ `ComplementaryArea/${ scope }` }>
+			<Animate type="slide-in" options={ { origin: 'left' } }>
+				{ () => <div className={ className }>{ children }</div> }
+			</Animate>
+		</Fill>
+	);
+}
+
+function ComplementaryArea( {
+	children,
+	className,
+	closeLabel = __( 'Close plugin' ),
+	complementaryAreaIdentifier,
+	header,
+	headerClassName,
+	icon,
+	isPinnable = true,
+	panelClassName,
+	scope,
+	smallScreenTitle,
+	title,
+	toggleShortcut,
+} ) {
+	const { isActive, isPinned } = useSelect(
+		( select ) => {
+			const { getActiveComplementaryArea, isItemPinned } = select(
+				'core/interface'
+			);
+			return {
+				isActive:
+					getActiveComplementaryArea( scope ) ===
+					complementaryAreaIdentifier,
+				isPinned: isItemPinned( scope, complementaryAreaIdentifier ),
+			};
+		},
+		[ complementaryAreaIdentifier, scope ]
+	);
+	const { enableComplementaryArea, disableComplementaryArea } = useDispatch(
+		'core/interface'
+	);
+	const { pinItem, unpinItem } = useDispatch( 'core/interface' );
+	return (
+		<>
+			{ isPinned && (
```

Comment:
The fact that `isPinned` is now outside of the `<PinnedItems />` fill created a regression as described in #28546 =.